projects/storefrontlib/src/cms-structure/page/slot/page-slot.component.ts

```diff
@@ -14,145 +14,184 @@ import {
-   * Use constructor(cmsService: CmsService, dynamicAttributeService: DynamicAttributeService, renderer: Renderer2, hostElement: ElementRef, config?: CmsConfig) instead
-   */
-  constructor(
-    cmsService: CmsService,
-    dynamicAttributeService: DynamicAttributeService,
-    renderer: Renderer2,
-    hostElement: ElementRef
-  );
+  /** Keeps track of the pending components that must be loaded for the page slot */
+  private pendingComponentCount = 0;
+
+  /** Tracks the last used position, in case the page slot is used dynamically */
+  private lastPosition: string;
   constructor(
     protected cmsService: CmsService,
     protected dynamicAttributeService: DynamicAttributeService,
     protected renderer: Renderer2,
-    protected hostElement: ElementRef,
-    protected config?: CmsConfig,
-    protected cdRef?: ChangeDetectorRef
+    protected elementRef: ElementRef,
+    protected config: CmsConfig,
+    protected cd: ChangeDetectorRef
   ) {}
 
   ngOnInit() {
     this.subscription.add(
-      this.components$.subscribe((components) => {
-        this.hasComponents = components && components.length > 0;
-        this.pendingComponentCount = components ? components.length : 0;
-        this.isPending = this.pendingComponentCount > 0;
-      })
+      this.slot$
+        .pipe(tap((slot) => this.decorate(slot)))
+        .subscribe((value) => (this.components = value?.components || []))
     );
   }
 
-  ngOnDestroy() {
-    this.subscription.unsubscribe();
+  protected decorate(slot: ContentSlotData): void {
+    let cls = this.class || '';
+
+    if (this.lastPosition && cls.indexOf(this.lastPosition) > -1) {
+      cls = cls.replace(this.lastPosition, '');
+    }
+    if (this.position$.value) {
+      cls += ` ${this.position$.value}`;
+      this.lastPosition = this.position$.value;
+    }
+
+    // host bindings
+    this.pending = slot?.components?.length || 0;
+    this.hasComponents = slot?.components?.length > 0;
+    if (cls && cls !== this.class) {
+      this.class = cls;
+    }
+
+    this.addSmartEditSlotClass(slot);
   }
 
   /**
-   * Is triggered when a component is added to the view.
-   * We use this information to dropthe `is-pending` class from the page slot
-   * when all nested components have been added.
+   * Sets the pending count for the page slot components. Once all pending components are
+   * loaded, the `isPending` flag is updated, so that the associated class can be updated
+   */
+  protected set pending(count: number) {
+    this.pendingComponentCount = count;
+    this.isPending = this.pendingComponentCount > 0;
+  }
+
+  protected get pending(): number {
+    return this.pendingComponentCount;
+  }
+
+  /**
+   * Toggles the css classes for the host element.
+   *
+   * @param cls the classname that should be added.
+   * @param forceAdd indicates that the class should be added or not regardless of the current classlist.
+   */
+  protected toggleStyleClass(cls: string, forceAdd?: boolean): void {
```

Comment:
Did you plan to reuse this logic?

Reply:
I thought yes, but indeed it's not needed as the BehaviourSubject is initialised explicitly.

Reply:
no! left over. thx.